### PR TITLE
Enable mentored bugs from bugzilla to be shown

### LINKED ISSUE #32: Handle mentored bugs for bugzilla
As of now, the bugs we are fetching from bugzilla are those labeled 'good-first-bug' as its tag, when I check on a few website linking to bugsahoy, most of the site mentioned that BugsAhoy are meant for mentored bugs so i think we may need to add capabilities for mentored bugs
TODO: find appropriate query to query mentored bugs

### PR DESCRIPTION
**Pull request by fiennyangeln**
-Refactor project/index.jsx
-Separate it into two routes, /projects/:project/mentored and
/projects/:project
-Move the AppBar into the project/index.jsx since it is similar
for both route
-Add mentored bugs filter using f1 and o1 properties

Closes #32

### CODE REVIEW

**Comment by helfi92** [comment]
How do I get this running? I am running https://github.com/mozilla-frontend-infra/bugzilla-graphql-gateway/pull/8 locally for the backend but when I click the "Show me mentored bugs/issues" button I get a network error.

screen shot 2018-07-23 at 10 52 31 am

**Comment by fiennyangeln** [comment]
@helfi92 Sorry forgot to write it down. Use this branch https://github.com/mozilla-frontend-infra/bugzilla-graphql-gateway/pull/8

**Comment by helfi92** [comment]
Are we able to do a query for mentored bugs in addition to the normal bugzilla query we usually do in the beginning with keyword good-first-bug? This would be a better experience than having the user click the filter icon then click the show mentored. I'm okay having to do two separate queries.

**Comment by helfi92** [comment]
What do you mean by "all the bugzilla query"?

**Comment by fiennyangeln** [comment]
i mean including the fetchMore we do after the initial query

**Comment by helfi92** [comment]
Sounds good. Let's give it a try and see how it goes.

**Comment by fiennyangeln** [comment]
@helfi92 found a way not to do more fetch, wdyt?

File: src/components/TasksTable/index.jsx [hunk 1]
```diff
@@ -335,11 +342,26 @@ export default class TasksTable extends Component {
...
+              component={Link}
+              to={{
+                pathname: url.includes('mentored')
+                  ? url.replace('mentored', '')
+                  : `${url}/mentored`.replace('//', '/'),
+                search: this.props.location.search,
+              }}>
+              SHOW{url.includes('mentored') ? ' GOOD FIRST ' : ' MENTORED '}
+              BUGS / ISSUES
+            </Button>
```
Comment (helfi92) [inline src/components/TasksTable/index.jsx]: I believe this button can be removed now.

File: src/views/Project/bugs.graphql [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+      component
+      lastChanged
+      id
+      keywords
+      summary
+      whiteboard
     }
   }
-}
+}
```
Comment (helfi92) [inline src/views/Project/bugs.graphql]: Add newline.

File: src/views/Project/index.jsx [hunk 3]
```diff
@@ -92,8 +92,23 @@ const tagReposMapping = repositories =>
     }) => ({
       fetchPolicy: 'network-only',
       variables: {
-        search: {
+        searchGoodFirst: {
```
Comment (helfi92) [inline src/views/Project/index.jsx]: Rename to `goodFirst`.

File: src/views/Project/index.jsx [hunk 4]
```diff
@@ -92,8 +92,23 @@ const tagReposMapping = repositories =>
...
+          ...(productsWithNoComponents(projects[project].products).length
+            ? { products: productsWithNoComponents(projects[project].products) }
+            : {
+                // otherwise, get only the first product and its components
+                // as component is not unique for product thus can't be merged
+                products: Object.keys(projects[project].products[0])[0],
+                components: Object.values(projects[project].products[0])[0],
+              }),
+        },
+        searchMentored: {
```
Comment (helfi92) [inline src/views/Project/index.jsx]: Rename to `mentored`.

File: src/components/TasksTable/index.jsx [hunk 5]
```diff
@@ -335,11 +342,26 @@ export default class TasksTable extends Component {
               ))}
             </TextField>
             <Button
+              className={classes.spacing}
```
Comment (helfi92) [inline src/components/TasksTable/index.jsx]: I think we can remove this class.

**Pull request merged**